src/expr.rs

```diff
@@ -674,3 +700,57 @@ fn rewrite_unary_op(context: &RewriteContext,
 
     Some(format!("{}{}", operator_str, subexpr))
 }
+
+fn rewrite_assign(context: &RewriteContext,
+                  lhs: &ast::Expr,
+                  rhs: &ast::Expr,
+                  op: Option<&ast::BinOp>,
+                  width: usize,
+                  offset: usize)
+                  -> Option<String> {
+    let operator_str = match op {
+        Some(op) => context.codemap.span_to_snippet(op.span).unwrap(),
+        None => "=".to_owned()
+    };
+
+    // 2 = " =".len()
+    let lhs_str = format!("{} {}",
+                          try_opt!(lhs.rewrite(context, width - operator_str.len() - 2, offset)),
+                          operator_str);
+
+    rewrite_assignment(&context, lhs_str, rhs, width, offset)
+}
+
+// The left hand side must contain everything up to, and including, the
+// assignment operator.
+pub fn rewrite_assignment<S: Into<String>>(context: &RewriteContext,
+                                           lhs: S,
+                                           ex: &ast::Expr,
+                                           width: usize,
+                                           offset: usize)
+                                           -> Option<String> {
+    let mut result = lhs.into();
+    result.push(' ');
```

Comment:
> I'd have thought we should consider a line split here, depending on the sub-expression lengths.

> Oh wait, you do, in which case I think this will lead to trailing whitespace.

> Oh wait again, you pop it in that case. Perhaps it is nice to only push if you need to, rather than sometimes popping.

- nrc